CMakeLists.txt

```diff
@@ -208,61 +222,125 @@ set(OPENCV_LIBS
-    COMMAND /bin/sh -c "if [ -f libmkl_avx2.so.1 ]; then patchelf --add-needed libmkl_core.so.1 libmkl_avx2.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_avx512.so.1 ]; then patchelf --add-needed libmkl_gnu_thread.so.1 libmkl_avx512.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_avx512.so.1 ]; then patchelf --add-needed libmkl_core.so.1 libmkl_avx512.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_vml_def.so.1 ]; then patchelf --add-needed libmkl_gnu_thread.so.1 libmkl_vml_def.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_vml_def.so.1 ]; then patchelf --add-needed libmkl_intel_thread.so.1 libmkl_vml_def.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_vml_def.so.1 ]; then patchelf --add-needed libmkl_core.so.1 libmkl_vml_def.so.1; fi"
-    COMMAND /bin/sh -c "if [ -f libmkl_intel_thread.so.1 ]; then patchelf --add-needed libmkl_intel_lp64.so.1 libmkl_intel_thread.so.1; fi"
-    COMMAND docker rm pytorch_backend_ptlib
-    COMMENT "Extracting pytorch and torchvision libraries and includes from ${TRITON_PYTORCH_DOCKER_IMAGE}"
-    VERBATIM
-  )
+  if(${RHEL_BUILD})
+    add_custom_command(
+      OUTPUT
+        ${PT_LIBS}
+        ${LIBTORCH_LIBS}
+        ${OPENCV_LIBS}
+        LICENSE.pytorch
+        include/torch
+      # TODO: TPRD-371 Support Torchvision for manylinux
+      #   include/torchvision
+      # COMMAND ${CMAKE_COMMAND} -E make_directory "include/torchvision"
+      COMMAND ${CMAKE_COMMAND} -E make_directory "include"
+      COMMAND docker pull ${TRITON_PYTORCH_DOCKER_IMAGE}
+      COMMAND docker rm pytorch_backend_ptlib || echo "error ignored..." || true
+      COMMAND docker create --name pytorch_backend_ptlib ${TRITON_PYTORCH_DOCKER_IMAGE}
+      # Transfer and install torch wheel which contains many of the target libraries
+      COMMAND docker cp pytorch_backend_ptlib:/opt/transfer/. /tmp/ && /bin/sh -c "pip install /tmp/torch-*.whl"
+      COMMAND /bin/sh -c "for i in ${LIBTORCH_LIBS_STR} ; do echo copying $i && docker cp -L pytorch_backend_ptlib:${LIBTORCH_PATH}/$i $i ; done"
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libc10.so libc10.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libc10_cuda.so libc10_cuda.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libtorch.so libtorch.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libtorch_cpu.so libtorch_cpu.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libtorch_cuda.so libtorch_cuda.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libtorch_cuda_linalg.so libtorch_cuda_linalg.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libtorch_global_deps.so libtorch_global_deps.so
+      COMMAND cp ${LIBTORCH_PATH}/python3.10/site-packages/torch/lib/libcaffe2_nvrtc.so libcaffe2_nvrtc.so
```

Comment:
Can we use `Python3_SITEARCH` as it described in documentation?
https://cmake.org/cmake/help/latest/module/FindPython3.html

Reply:
if the list of the libs build depend on between Debian and RedHat  is equal. 
Ten only portion evaluation should resolve the path that will be used for copy, isn't?

Reply:
If I am understanding you correctly, you would like there to be no if/else clause since they share similar dependencies? This is something I wanted to avoid as well, however, the issue here is that for RHEL the files will be local (because it's where we install the torch wheel), whereas, for Debian, the image we use has the wheel pre-installed so we copy it out from the image.

That said, I default `${LIBTORCH_PATH}` to `/usr/local/lib`, so I have the opportunity to clean up some of the debian paths below by prefixing with `${LIBTORCH_PATH}`. I'll have to look into `Python3_SITEARCH` to see how it could factor in.

Reply:
Tried to leverage `FindPython` and oddly most of the results variables are blank:
```
Python3_STDARCH
Python3_STDLIB
Python3_SITELIB
Python3_SITEARCH
Python3_INCLUDE_DIRS /opt/_internal/cpython-3.10.13/include/python3.10 <-- only value retrieved
Python3_LIBRARIES
Python3_LIBRARY_DIRS
```